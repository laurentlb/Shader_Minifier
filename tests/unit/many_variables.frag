// Lots of identifiers to force Shader Minifier to use 2-char idents.

int fct(float a, float b, float c, float d, float e, float f,
        float g, float h, float i, float j, float k, float l) {
  float m = a, n = b, o = c, p = d, q = e, r = f;
  int v1 = 1, v2 = 2, v3 = 3, v4 = 4, v5 = 5, v6 = 6, v7 = 7, v8 = 8; 
  float s = g, t = h, u = i, v = j, w = k, x = l;
  int a1 = 1, a2 = 2, a3 = 3, a4 = 4, a5 = 5, a6 = 6, a7 = 7, a8 = 8;
  float y = 0.;
  int b1 = 1, b2 = 2, b3 = 3, b4 = 4, b5 = 5, b6 = 6, b7 = 7, b8 = 8;  
  float z = 0.;
  int c1 = 1, c2 = 2, c3 = 3, c4 = 4, c5 = 5, c6 = 6, c7 = 7, c8 = 8;
  return v8 + a8 + b8 + c8;
}
