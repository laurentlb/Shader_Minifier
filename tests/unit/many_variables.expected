/* File generated with Shader Minifier 1.1.6
 * http://www.ctrl-alt-test.fr
 */

// tests/unit/many_variables.frag
"int t(float t,float o,float l,float f,float a,float n,float i,float r,float u,float e,float Z,float Y)"
 "{"
   "float X=t,W=o,V=l,U=f,T=a,S=n;"
   "int R=1,Q=2,P=3,O=4,N=5,M=6,L=7,K=8;"
   "float J=i,I=r,H=u,G=e,F=Z,E=Y;"
   "int D=1,C=2,B=3,A=4,z=5,y=6,x=7,w=8;"
   "float v=0.;"
   "int s=1,q=2,p=3,m=4,k=5,j=6,h=7,g=8;"
   "float d=0.;"
   "int c=1,b=2,at=3,ab=4,ac=5,ad=6,ag=7,ah=8;"
   "return K+w+g+ah;"
 "}",
